test/helpers/test.go

```diff
@@ -832,17 +938,50 @@ func (t *TestOptions) ExpectThatIssueIsAssignedToUser(provider gits.GitProvider,
 	return RetryExponentialBackoff(TimeoutProwActionWait, f)
 }
 
-// AttemptToLGTMOwnPR return an error if the /lgtm fails to add the lgtm label to PR
-func (t *TestOptions) AttemptToLGTMOwnPR(provider gits.GitProvider, owner string, repo string) error {
+// MostRecentOpenPullRequestForOwnerAndRepo returns the most recently opened pull request for a given owner/repo. If
+// there aren't any open PRs, it will return nil.
+func (t *TestOptions) MostRecentOpenPullRequestForOwnerAndRepo(provider gits.GitProvider, owner string, repo string) (*gits.GitPullRequest, error) {
 	pullRequests, err := provider.ListOpenPullRequests(owner, repo)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	if len(pullRequests) < 1 {
-		return fmt.Errorf("no open pull requests found for %s/%s", owner, repo)
+		return nil, fmt.Errorf("no open pull requests found for %s/%s", owner, repo)
 	}
+	sort.SliceStable(pullRequests, func(i, j int) bool {
+		iNum := 0
+		jNum := 0
+		if pullRequests[i].Number != nil {
+			iNum = *pullRequests[i].Number
+		}
+		if pullRequests[j].Number != nil {
+			jNum = *pullRequests[j].Number
+		}
+		return iNum > jNum
+	})
 
-	err = provider.AddPRComment(pullRequests[0], "/lgtm")
+	// The first element in the slice is the open PR with the highest number.
+	return pullRequests[0], nil
+}
+
+// ApprovePR attempts to /approve a PR with the given approver git provider, then verify the label is there with the default provider
+func (t *TestOptions) ApprovePR(defaultProvider gits.GitProvider, approverProvider gits.GitProvider, pullRequest *gits.GitPullRequest) error {
+	err := approverProvider.AddPRComment(pullRequest, "/approve")
+	if err != nil {
+		return err
+	}
+
+	return t.ExpectThatPRHasLabel(defaultProvider, *pullRequest.Number, pullRequest.Owner, pullRequest.Repo, "approved")
```

Comment:
Is there a timeout or retry on this per default? There is a delay between adding a comment and Prow converting this into a label.

Reply:
Yup, https://github.com/jenkins-x/bdd-jx/blob/9649c357d1e03889c2de1b1ef63b74cab7eaaac0/test/helpers/test.go#L1084 in `ExpectThatPRHasLabel`.